src/contract.ts

```diff
@@ -46,7 +49,7 @@ export class AbstractContract {
 
   get lockingScript(): Script {
     let lsASM = this.scriptedConstructor.toASM();
-    if (this._dataPart !== undefined && this._dataPart !== null) {
+    if (this._dataPart) {
```

Comment:
this._dataPart could be empty string, in which case, `opreturn` should also be added.

Reply:
Could change to:
if (this._dataPart) {
   lsASM += ` OP_RETURN ${this._dataPart}`;
} else if (this._dataPart.trim() === '') {
  lsASM += ` OP_RETURN`;
}

Reply:
if (typeof this._dataPart === 'string') {
      const dp = this._dataPart.trim();
      if (dp) {
        lsASM += ` OP_RETURN ${dp}`;
      } else {
        lsASM += ` OP_RETURN`;  // note there is no space after op_return
      }
    }

Reply:
f6181c8e687b92b652820e621b113ab6837ec23c